src/DataModel/IMetadataProvider.ts

```diff
@@ -0,0 +1,30 @@
+import {IMetadataDescription} from "./IMetadataDescription";
+import "whatwg-fetch";
+import {IData} from "./IData";
+
+/**
+ * @interface Describes an abstract meta-data providing facility which translates from raw @link Response to an abstract data model.
+ */
+export interface IMetadataProvider
+{
+    /**
+     * Gets supported media types.
+     * @returns {Array<string>}
+     */
+    supportedMediaTypes: Array<string>;
+
+    /**
+     * Parses a given raw response.
+     * @param {Response} response Raw fetch response holding data to be parsed.
+     * @returns {Promise<IData>}
+     */
+    parse(response: Response): Promise<IData>;
```

Comment:
Sure, but I think that parsing is just part of what `IMetadataProvider` does, is all.